Product list filters NEXT-8591 #880

NTWEXO opened the pull request:
### 1. Why is this change necessary?
Fix for issue #880 

### 2. What does this change do, exactly?
Allows the filtering of products on the product list page in the admin panel.

The filter sidebar item is created as its own component and can be used on any list page with by simply passing an array of what filter options is wanted.

### 3. Describe each step to reproduce the issue or behaviour.
Go to product listing page in the administration panel, before change is was not possible to filter in the result, after change it is.

### 4. Please link to the relevant issues (if any).
#880 

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

schliflo commented:
By the way I think your tests are broken. Travis build for PHP 5.6 failed in the first run: https://travis-ci.org/shopware/shopware/builds/197600294
I had to amend and force push without changing code to re-trigger travis. Build passed the second time around...

wolv-dev commented:
Ticket opened!
https://issues.shopware.com/issues/SW-18547
Something like this shouldn't be closed, cause the code quality should be maintained.

NTWEXO committed:
fix switch returning false when unselected #880

Ion-manden committed:
NEXT-8591 filter products #880

Moved sidebar filter to its own component and made is configurable and usable in any list
changed for loops to foreach loop #8591

cs fix missing semicollon #8591

fix reactivity in range filter and text input #8591

cs fix #8591

pweyck commented:
Hello @Ion-manden, could you sign the CLA?

Ion-manden commented:
@pweyck accepted it a while back, did it not go through?
